lib/active_publisher/async.rb

```diff
@@ -0,0 +1,29 @@
+module ActivePublisher
+  # Publish a message asynchronously to RabbitMQ.
+  #
+  # Asynchronous is designed to do two things:
+  # 1. Introduce the idea of a durable retry should the RabbitMQ connection disconnect.
+  # 2. Provide a higher-level pattern for fire-and-forget publishing.
+  #
+  # @param [String] route The routing key to use for this message.
+  # @param [String] payload The message you are sending. Should already be encoded as a string.
+  # @param [String] exchange The exchange you want to publish to.
+  # @param [Hash] options hash to set message parameters (e.g. headers).
+  def self.publish_async(route, payload, exchange_name, options = {})
+    ::ActivePublisher::Async.publisher_adapter.publish(route, payload, exchange_name, options)
+  end
+
+  module Async
+    def self.publisher_adapter
+      @publisher_adapter ||= case ::ActivePublisher.configuration.async_publisher
```

Comment:
I see two issues with providing an abstraction:
* We have to provide abstraction for configuring the adapter. How do you change the max messages it will hold before it starts rejecting them? Does it use publisher confirms?
* If someone creates their own adapter we have to know about it and add it to our abstraction.

Rather than provide an abstraction I would prefer to document it clearly in our `README.md` and provide an example of how to use the in memory adapter if they want to do that.